mllib/src/main/scala/org/apache/spark/ml/ann/Layer.scala

```diff
@@ -463,7 +479,7 @@ private[ml] class FeedForwardModel private(
   private var outputs: Array[BDM[Double]] = null
   private var deltas: Array[BDM[Double]] = null
 
-  override def forward(data: BDM[Double]): Array[BDM[Double]] = {
+  override def forward(data: BDM[Double], containsLastLayer: Boolean): Array[BDM[Double]] = {
     // Initialize output arrays for all layers. Special treatment for InPlace
```

Comment:
The last layer is always `softmax`, add the `containsLastLayer` parameter, when `true` the forward computing will contains last layer, otherwise not. The parameter is used when we need `rawPrediction`, the last layer `softmax` should discard.